compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -3747,6 +3784,30 @@ private boolean isAssignableToUnionType(BType source, BType target, Set<TypePair
         return true;
     }
 
+    private boolean isUnionDistinctError(Set<BType> memberTypesSet) {
+        List<BType> memberTypes = new ArrayList<>(memberTypesSet);
+        int size = memberTypes.size();
+        for (int i = 0; i < size - 1; i++) {
+            BErrorType t1 = (BErrorType) memberTypes.get(i);
+            BErrorType t2 = (BErrorType) memberTypes.get(i + 1);
+            Set<BTypeIdSet.BTypeId> t1TypeIds = t1.typeIdSet.getAll();
+            Set<BTypeIdSet.BTypeId> t2TypeIds = t2.typeIdSet.getAll();
+            if (!t1TypeIds.containsAll(t2TypeIds) || !t2TypeIds.containsAll(t1TypeIds)) {
```

Comment:
I believe we can just use equals
`if (!t1TypeIds.equals(t2TypeIds)) {
 return false;
}
`

Reply:
Changed.